### PR TITLE
libsixel: new package

### PR DESCRIPTION
**Pull request by taliaferro**
Libsixel is a library for dealing with sixel graphics -- a way of representing images as terminal escape sequences, which can be displayed by some terminal emulators, like iTerm2. The code, and additional information, can be found at [saitoha/libsixel](https://github.com/saitoha/libsixel).\
This package produces the `libsixel` library, and also an `img2sixel` command-line tool which can convert images to the sixel format.

I've got this to build successfully with the `+jpeg` and `+png` variants (which I set as the default) but I added variants for most of the flags supported by the `configure` script for the project.

I also didn't include the flag that also builds the Python bindings, but that could be added in the future -- either in this package or in a separate, dependent `py-libsixel` package?

### CODE REVIEW

File: var/spack/repos/builtin/packages/libsixel/package.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+    variant("gdk-pixbuf2", default=False, description="build with gdk-pixbuf2")
+    depends_on("gdk-pixbuf", when="+gdk-pixbuf2")
+
+    variant("jpeg", default=True, description="build with libjpeg")
+    depends_on("libjpeg", when="+jpeg")
+
+    variant("png", default=True, description="build with libpng")
+    depends_on("libpng", when="+png")
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/libsixel/package.py]:
```suggestion
    variant("curl", default=False, description="build with libcurl")
    variant("gd", default=False, description="build with libgd")
    variant("gdk-pixbuf2", default=False, description="build with gdk-pixbuf2")
    variant("jpeg", default=True, description="build with libjpeg")
    variant("png", default=True, description="build with libpng")

    depends_on("curl", when="+curl")
    depends_on("gdk-pixbuf", when="+gdk-pixbuf2")
    depends_on("libgd", when="+gd")
    depends_on("libjpeg", when="+jpeg")
    depends_on("libpng", when="+png")
```

Just a quick nitpick. It makes it easier to review packages if they follow the common format of
```
1. versions
2. variants
3. dependencies
```

**Review by alecbcs — CHANGES_REQUESTED** [review]
Nice job on your first package @taliaferro! Just a quick nitpick about variant/dependency formatting.

File: var/spack/repos/builtin/packages/libsixel/package.py (hunk 1 above)
Comment (taliaferro) [reply var/spack/repos/builtin/packages/libsixel/package.py]: I've reordered things so variants and dependencies are grouped with each other, and formatted it so it should pass the style checks. Do you have any other recommendations?

File: var/spack/repos/builtin/packages/libsixel/package.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Libsixel(MesonPackage):
+    """
+    This package provides encoder/decoder implementation for DEC SIXEL graphics, and some converter programs.
```
Comment (alecbcs) [inline var/spack/repos/builtin/packages/libsixel/package.py]: @taliaferro can you break up this line into to two to fix the failing style check. (The line is >100 characters long.)

Comment (taliaferro) [reply var/spack/repos/builtin/packages/libsixel/package.py]: Sure, thanks for the reminder!

**Review by alecbcs — APPROVED** [review]
Looks good to me. Thanks @taliaferro!

**Pull request merged**